src/modules/simulator/simulator_mavlink.cpp

```diff
@@ -839,7 +845,8 @@ void Simulator::pollForMAVLinkMessages(bool publish)
 				for (int i = 0; i < len; ++i) {
 					if (mavlink_parse_char(MAVLINK_COMM_1, serial_buf[i], &msg, &serial_status)) {
 						// have a message, handle it
-						handle_message(&msg, true);
+						_publish = true;
```

Comment:
Since you're overriding the `_publish` state here, it should be restored after the `handle_message` call.